### ISSUE TITLE
Copy Constructor does not utilize isSetXXX methods when they are generated.

### ISSUE THREAD

**Issue by fericit-bostan** [sabomichal/immutable-xjc#8]
I am utilizing -immutable, -imm-builder and -imm-cc configuration flags. The Builder's copy constructor is generated as are all of the isSetXXX methods. The problem is that the copy constructor is invoking all of the getXXX methods of the supplied object rather than performing a copy of the fields. The side effect is that if the supplied Column's field has not been set and has a default value specified for a field  then the default value is copied over rather than leaving the field's value as null. 

One of two actions should occur: 
1 - the copy operation should copy the fields value and not invoke the getXXX methods.
2 - if the isSetXXX methods are generated then the copy operation should be as follows:
if(o.isSetXXX()) {
  this.XXX = o.getXXX();
}

In my opinion, option 1 is the easier solution to implement and would still honor the contract of the isSetXXX methods.

By the way, great job on this implementation. I find it very clean and I appreciate the clean, generated code very much. A job well done!!!

**Comment by sabomichal** [comment]
Thanks for filing the issue, you are right, it will be fixed.

**Comment by fericit-bostan** [comment]
When do you think that you will release the next version of the plugin? I'd really like to pickup the changes.

Thanks for giving the issue attention.

**Comment by sabomichal** [comment]
Since you were giving this repo a star, I'll try to do it ASAP. :)

**Comment by sabomichal** [comment]
Check if current HEAD is working for you, I will then do the release. Nevertheless isSetXXX methods are not generated by default (I think you are leveraging the jaxb:bindings), so they were no use. I used the first proposed solution, however just for the base class. Assignment of superclass fields can be done just by calling the getters.

**Comment by sabomichal** [comment]
And as I see the commit was not interpreted correctly by git, so it looks like the whole file was changed. :-/

**Comment by fericit-bostan** [comment]
I just cloned the repo and installed the build. I ran my project and the generated code (copy constructor) looks good. 

I didn't think the isSetXXX was a good solution to the issue, and I had forgotten that I had asked jaxb to generate those methods, so you are correct in that it would not have worked. This is the better solution to the problem.

Thank you for providing a fix for this so quickly. I'll look for the release.
